### DISCUSSION TITLE
Blazor Server -- updated to latest package, how to initialize...

### DISCUSSION THREAD

**Discussion by kfinke70** [IvanJosipovic/BlazorApplicationInsights#247] [Q&A] [answered]
I have a .NET 7 Blazor server app, still using Startup.cs.   I had been using the prior version of this package fine.  I upgraded, and now I'm not sure how to use the new initialization.  Part of my Startup is shown:
`        public void ConfigureServices(IServiceCollection services)
        {
            RegisterCognito(services);

            services.AddCloudRoleNameInitializer(Configuration["ApplicationInsights:RoleName"]);
            services.AddApplicationInsightsTelemetry();

            services.AddBlazorApplicationInsights(
                appInsights =>
                {
                    appInsights.ConnectionString = Configuration["ApplicationInsights:ConnectionString"];
                },
                    async ai =>
                    {
                        try
                        {
                            {
                                var telemetryItem = new TelemetryItem()
                                {
                                    Tags = new Dictionary<string, object?>()
                            {
                                { "ai.cloud.role", "SPA" },
                                { "ai.cloud.roleInstance", "Blazor Wasm" },
                            }
                                };

                                await ai.AddTelemetryInitializer(telemetryItem);
                            };
                        }
                        catch (Exception e)
                        {
                            Console.WriteLine("Could not initialize ApplicationInsights");
                            Console.WriteLine(e);
                        }
                    }
            );`

I get an error from DI saying:
Lifetime: Singleton ImplementationType: BlazorApplicationInsights.ApplicationInsights': Cannot consume scoped service 'Microsoft.JSInterop.IJSRuntime' from singleton 'BlazorApplicationInsights.Interfaces.IApplicationInsights'.)

**Comment by IvanJosipovic** [comment, marked as answer]
I converted this to an issue, https://github.com/IvanJosipovic/BlazorApplicationInsights/issues/248
